frontend/public/components/edit-yaml.jsx

```diff
@@ -369,14 +439,59 @@ export const EditYAML_ = connect(stateToProps)(
               managedResourceSaveModal({
                 kind: obj.kind,
                 resource: obj,
-                onSubmit: () => this.updateYAML(obj, model, newNamespace, newName),
+                onSubmit: () => this.updateYAML(obj, getModel(models, obj), newNamespace, newName),
                 owner,
               });
               return;
             }
           }
+          this.updateYAML(obj, getModel(models, obj), newNamespace, newName);
+        }
 
-          this.updateYAML(obj, model, newNamespace, newName);
+        saveAll() {
+          const { t } = this.props;
+          let objs;
+          let hasErrors = false;
+          this.setState({ errors: null }, () => {
+            try {
+              objs = safeLoadAll(this.getEditor().getValue());
+            } catch (e) {
+              this.handleError(t('public~Error parsing YAML: {{e}}', { e }));
+              return;
+            }
+            if (objs.length === 1) {
+              this.save();
+              return;
+            } else if (objs.length === 0) {
+              return;
+            }
+            //Run through client side validation for all resources
+            objs.forEach((obj) => {
+              const validationError = this.validateYAML(obj);
+              if (validationError) {
+                hasErrors = true;
+                this.handleErrors(obj, validationError);
+              }
+            });
+            if (!hasErrors) {
+              //Check for duplicate name/kinds
+              const uniqueEntries = _.uniqBy(objs, (obj) =>
+                [
+                  obj.metadata.name,
+                  obj.metadata.namespace,
+                  obj.kind,
+                  groupVersionFor(obj.apiVersion).group,
+                ].join('-'),
```

Comment:
`-` isn't a good separate here since it's allowed in names. You would still have false positives for

name: foo-bar
namespace: baz

and

name: foo
namespace: bar-baz

Use something like `~` which isn't allowed in names.